source/common/http/conn_manager_impl.cc

```diff
@@ -1189,9 +1199,10 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
 
   // Now actually encode via the codec.
   stream_info_.onFirstDownstreamTxByteSent();
-  response_encoder_->encodeHeaders(
-      headers,
-      encoding_headers_only_ || (end_stream && continue_data_entry == encoder_filters_.end()));
+  const bool encoder_end_stream =
+      encoding_headers_only_ || (end_stream && continue_data_entry == encoder_filters_.end());
+  response_encoder_->encodeHeaders(headers, encoder_end_stream);
+  response_encoder_end_stream_ = encoder_end_stream;
```

Comment:
I'm a a bit lost on why we need a stream level `response_encoder_end_stream_`. Why can't this be the inverse of decode and you just set the `end_stream_` boolean on the filter? I think you actually want to do this at the filter level and not at the codec level which I think you are doing?

Reply:
You are right.  The filter level end_stream is enough.  This flag is removed